kernel/mellanox_ethernet/src/command_queue.rs

```diff
@@ -0,0 +1,740 @@
+            CommandDeliveryStatus::BadOutputPointer
+        } else if status == 5 {
+            CommandDeliveryStatus::BadInputPointer
+        } else if status == 6 {
+            CommandDeliveryStatus::InternalErr
+        } else if status == 7 {
+            CommandDeliveryStatus::InputLenErr
+        } else if status == 8 {
+            CommandDeliveryStatus::OutputLenErr
+        } else if status == 9 {
+            CommandDeliveryStatus::ReservedNotZero
+        } else if status == 10 {
+            CommandDeliveryStatus::BadCommandType
+        } else {
+            CommandDeliveryStatus::Unknown
+        }
+    }
+
+    /// Sets the ownership bit so that HW can take control of the command entry
+    pub fn change_ownership_to_hw(&mut self) {
+        let ownership = self.token_signature_status_own.read().get() | 0x1;
+        self.token_signature_status_own.write(U32::new(ownership));
+    }
+
+    /// Returns true if the command is currently under the ownership of HW (SW should not touch the fields).
+    pub fn owned_by_hw(&self) -> bool {
+        self.token_signature_status_own.read().get().get_bit(0)
+    }
+
+    /// Returns the status of command execution.
+    pub fn get_return_status(&self) -> CommandReturnStatus {
+        let (status, _syndrome, _, _) = self.get_output_inline_data();
+
+        if status == 0x0 {
+            CommandReturnStatus::OK
+        } else if status == 0x1 {
+            CommandReturnStatus::InternalError
+        } else if status == 0x2 {
+            CommandReturnStatus::BadOp
+        } else if status == 0x3 {
+            CommandReturnStatus::BadParam
+        } else if status == 0x4 {
+            CommandReturnStatus::BadSysState
+        } else if status == 0x5 {
+            CommandReturnStatus::BadResource
+        } else if status == 0x6 {
+            CommandReturnStatus::ResourceBusy
+        } else if status == 0x8 {
+            CommandReturnStatus::ExceedLim
+        } else if status == 0x9 {
+            CommandReturnStatus::BadResState
+        } else if status == 0xA {
+            CommandReturnStatus::BadIndex
+        } else if status == 0xF {
+            CommandReturnStatus::NoResources
+        } else if status == 0x50 {
+            CommandReturnStatus::BadInputLen
+        } else if status == 0x51 {
+            CommandReturnStatus::BadOutputLen
+        } else if status == 0x10 {
+            CommandReturnStatus::BadResourceState
+        } else if status == 0x30 {
+            CommandReturnStatus::BadPkt
+        } else if status == 0x40 {
+            CommandReturnStatus::BadSize
+        } else {
+            CommandReturnStatus::Unknown
+        }
+    }
+}
+
+
+#[derive(FromBytes)]
+#[repr(C)]
+/// Layout of mailbox used to pass extra input and output command data that doesn't fit into the command entry.
+struct CommandInterfaceMailbox {
+    /// Data in the mailbox
+    mailbox_data:           Volatile<[u8; 512]>,
+    _padding:               ReadOnly<[u8; 48]>,
```

Comment:
does this `_padding` field actually need to be read from? If not, just specify it as a plain array like we usually do: `_padding: [u8; 48]`